### PR TITLE
Add User assign and summary view for create role flow.

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: modules/theme/src/themes/default/elements/segment.variables [hunk 1]
```diff
@@ -40,3 +40,10 @@
 @assignedRolesListIconMarginLeft: 0.3em;
 @InfoLabelBackgroundColor: #b5b7b7;
 @InfoLabelBorderColor: #b5b7b7;
+
+/*************************************
+    Create Role Summary Variation
+*************************************/
+
+@RoleSummaryUserListBackgroundColor: #f7f7f7;
+@RoleSummaryUserListBackgroundColor: #efefef;
```
Comment (thanujalk) [inline modules/theme/src/themes/default/elements/segment.variables]: add new line at the EOF

Comment (DimalChandrasiri) [reply modules/theme/src/themes/default/elements/segment.variables]: Fixed with commit 63c6790

File: apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { Grid, Label } from "semantic-ui-react";
+import { UserAvatar } from "@wso2is/react-components";
+
+interface AddUserWizardSummaryProps {
+    summary: any;
+    triggerSubmit: boolean;
+    onSubmit: (application: any) => void;
+}
+
+export const CreateRoleSummary: FunctionComponent<AddUserWizardSummaryProps> = (props: AddUserWizardSummaryProps): ReactElement => {
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx]: Fix character length

File: apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                                    }
+                                </Label.Group>
+                            </Grid.Column>
+                        </Grid.Row>
+                    )
+                    : null
+            }
+        </Grid>
+    )
+}
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx]: Missing newline at EOF

File: apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                                        summary.RoleUserList
+                                            .map((user, index) => (
+                                                <div key={ index } className="role-summary-user">
+                                                    <UserAvatar
+                                                        name={ user.userName }
+                                                        size="mini"
+                                                        floated="left"
+                                                        image={ user.profileUrl }
+                                                    />
+                                                    {user.userName}
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx]:
```suggestion
                                                    { user.userName }
```

File: apps/admin-portal/src/components/users/role-wizard/role-user-assign.tsx [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                                                    <UserAvatar
+                                                        name={ user.userName }
+                                                        size="mini"
+                                                        floated="left"
+                                                        image={ user.profileUrl }
+                                                    />
+                                                     { user.userName }
+                                                     <Icon
+                                                         name="add"
+                                                     />
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-user-assign.tsx]:
```suggestion
                                                     <Icon name="add" />
```

File: apps/admin-portal/src/components/users/role-wizard/role-wizard.tsx [hunk 6]
```diff
@@ -40,6 +42,18 @@ interface CreateRoleProps {
...
 
+/**
+ * Enum for wizard steps form types.
+ * @readonly
+ * @enum {string}
+ */
+enum WizardStepsFormTypes {
+    BASIC_DETAILS = "BasicDetails",
+    PERM_LIST= "PermissionList",
+    USER_LIST= "RoleUserList",
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-wizard.tsx]:
```suggestion
    PERM_LIST = "PermissionList",
    USER_LIST = "RoleUserList",
```

File: apps/admin-portal/src/components/users/role-wizard/role-wizard.tsx [hunk 7]
```diff
@@ -64,22 +78,122 @@ export const CreateRoleWizard: FunctionComponent<CreateRoleProps> = (props: Crea
...
     /**
      * Method to handle create role action when create role wizard finish action is triggered.
      * 
      * @param basicData - basic data required to create role.
      * @param permissions - permissions selected for the created role.
      */
-    const addRole = (basicData: any, permissions: string[]): void => {
+    const addRole = (basicData: any): void => {
+        let members: CreateRoleMemberInterface[] = [];
+        if (basicData.RoleUserList.length > 0) {
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/users/role-wizard/role-wizard.tsx]:
```suggestion
        let members: CreateRoleMemberInterface[] = [];

        if (basicData.RoleUserList.length > 0) {
```

File: modules/theme/src/themes/default/elements/segment.overrides [hunk 8]
```diff
@@ -122,3 +142,21 @@
...
+    background-color: #f7f7f7;
+    border-radius: 100px;
+    align-items: center;
+    padding: 0 1em 0 0;
+    border: 0.5px solid @RoleSummaryUserListBackgroundColor;
+
+    .circular {
+        margin-right: 0.5em !important;
+    }
+}
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/elements/segment.overrides]: Missing newline at EOF

```suggestion
}

```

File: apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx (hunk 2 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/components/users/role-wizard/role-sumary.tsx]: Fixed with commit 4a29d6a

**Pull request merged**